deploy/AllocatorInit.sol

```diff
@@ -0,0 +1,278 @@
+
+        require(SwapperLike(networkInstance.swapper).roles()  == sharedInstance.roles,   "AllocatorInit/swapper-roles-mismatch");
+        require(SwapperLike(networkInstance.swapper).ilk()    == ilk,                    "AllocatorInit/swapper-ilk-mismatch");
+        require(SwapperLike(networkInstance.swapper).buffer() == networkInstance.buffer, "AllocatorInit/swapper-buffer-mismatch");
+
+        require(DepositorUniV3Like(networkInstance.depositorUniV3).roles()  == sharedInstance.roles,   "AllocatorInit/depositorUniV3-roles-mismatch");
+        require(DepositorUniV3Like(networkInstance.depositorUniV3).ilk()    == ilk,                    "AllocatorInit/depositorUniV3-ilk-mismatch");
+        require(DepositorUniV3Like(networkInstance.depositorUniV3).buffer() == networkInstance.buffer, "AllocatorInit/depositorUniV3-buffer-mismatch");
+
+        require(StableSwapperLike(networkInstance.stableSwapper).swapper()                 == networkInstance.swapper,        "AllocatorInit/stableSwapper-swapper-mismatch");
+        require(StableDepositorUniV3Like(networkInstance.stableDepositorUniV3).depositor() == networkInstance.depositorUniV3, "AllocatorInit/stableDepositorUniV3-depositorUniV3-mismatch");
+
+        require(ConduitMoverLike(networkInstance.conduitMover).ilk()    == ilk,                    "AllocatorInit/conduitMover-ilk-mismatch");
+        require(ConduitMoverLike(networkInstance.conduitMover).buffer() == networkInstance.buffer, "AllocatorInit/conduitMover-buffer-mismatch");
+
+        // Onboard the ilk
+        dss.vat.init(ilk);
+        dss.jug.init(ilk);
+
+        require(cfg.debtCeiling < WAD, "AllocatorInit/incorrect-ilk-line-precision");
+        dss.vat.file(ilk, "line", cfg.debtCeiling * RAD);
+        dss.vat.file("Line", dss.vat.Line() + cfg.debtCeiling * RAD);
+
+        dss.spotter.file(ilk, "pip", sharedInstance.oracle);
+        dss.spotter.file(ilk, "mat", RAY);
+        dss.spotter.poke(ilk);
+
+        // Add buffer to registry
+        RegistryLike(sharedInstance.registry).file(ilk, "buffer", networkInstance.buffer);
+
+        // Initiate the allocator vault
+        dss.vat.slip(ilk, networkInstance.vault, int256(1_000_000 * WAD));
+        dss.vat.grab(ilk, networkInstance.vault, networkInstance.vault, address(0), int256(1_000_000 * WAD), 0);
+
+        VaultLike(networkInstance.vault).file("jug", address(dss.jug));
+
+        // Allow vault and funnels to pull funds from the buffer
+        BufferLike(networkInstance.buffer).approve(VaultLike(networkInstance.vault).nst(), networkInstance.vault, type(uint256).max);
+        for(uint256 i = 0; i < cfg.swapGems.length; i++) {
+            BufferLike(networkInstance.buffer).approve(cfg.swapGems[i], networkInstance.swapper, type(uint256).max);
+        }
+        for(uint256 i = 0; i < cfg.depositGems.length; i++) {
+            BufferLike(networkInstance.buffer).approve(cfg.depositGems[i], networkInstance.depositorUniV3, type(uint256).max);
```

Comment:
Maybe nst can always be approved for both the swapper and the depositor, without requiring the nst token to be present in the `depositGems` and `swapGems` arrays. Would make sense given that term "gem" generally doesn't refer to dai/nst. Alternatively, we could rename the arrays `depositTokens` and `swapTokens`.

Reply:
I think there might be a possibility that nst would just be moved to a conduit without the need for swaps/deposits, and maybe the subdao wouldn't want unnecessary approvals to the swap and deposit funnels. So not seeing a big harm in requiring nst to be part of the tokens list.

I changed to `swapTokens` and `depositTokens`.